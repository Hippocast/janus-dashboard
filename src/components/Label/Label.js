import React from 'react'
import PropTypes from 'prop-types'

import './Label.css'

const propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.string
  ]).isRequired,
  htmlFor: PropTypes.string
}

const Label = ({ children, htmlFor }) => (
  <label className='j-label' htmlFor={htmlFor}>
    { children }
  </label>
)

Label.propTypes = propTypes

export default Label
