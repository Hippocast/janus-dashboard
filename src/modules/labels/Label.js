import React from 'react';
import PropTypes from 'prop-types';

import './Label.css';

const propTypes = {
    children: PropTypes.oneOfType([
        PropTypes.node,
        PropTypes.string,
    ]).isRequired,
};

const Label = ({ children }) => (
    <div className="j-label">
        { children }
    </div>
);

Label.propTypes = propTypes;

export default Label;
