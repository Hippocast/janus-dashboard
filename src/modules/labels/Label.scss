@import 'index.scss';

.j-label {
    font-size: $hf-font-size-base;
	color: $hf-color-black;
    line-height: $hf-line-height;
}
