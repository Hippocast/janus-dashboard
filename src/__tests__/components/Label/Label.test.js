import React from 'react';
import renderer from 'react-test-renderer';

import Label from '../../../components/Label/Label';

describe('Label component', () => {
    const renderWithRequiredProps = props => renderer.create(
        <Label
            {...props}
        />
    ).toJSON();

    it('renders correctly', () => {
        const passedProps = {
            children: 'mock-text',
            htmlFor: 'mock-target',
        };
        const tree = renderWithRequiredProps(passedProps);

        expect(tree).toMatchSnapshot();
    });

    it('renders with single child, if `children` is not a String', () => {
        const passedProps = {
            children: <p>mock-child</p>,
        };
        const tree = renderWithRequiredProps(passedProps);

        expect(tree).toMatchSnapshot();
    });

    it('renders with all children,  if `children` is not a String', () => {
        const passedProps = {
            children: [
                <p key="1">mock-child-1</p>,
                <p key="2">mock-child-2</p>,
                <p key="3">mock-child-3</p>
            ],
        };
        const tree = renderWithRequiredProps(passedProps);

        expect(tree).toMatchSnapshot();
    });
});
