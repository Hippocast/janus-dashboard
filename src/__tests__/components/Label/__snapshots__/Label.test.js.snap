// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label component renders correctly 1`] = `
<label
  className="j-label"
  htmlFor="mock-target"
>
  mock-text
</label>
`;

exports[`Label component renders with all children,  if \`children\` is not a String 1`] = `
<label
  className="j-label"
  htmlFor={undefined}
>
  <p>
    mock-child-1
  </p>
  <p>
    mock-child-2
  </p>
  <p>
    mock-child-3
  </p>
</label>
`;

exports[`Label component renders with single child, if \`children\` is not a String 1`] = `
<label
  className="j-label"
  htmlFor={undefined}
>
  <p>
    mock-child
  </p>
</label>
`;
